src/compressed_tensors/quantization/lifecycle/forward.py

```diff
@@ -60,9 +58,20 @@ def fake_quantize(
     bit_range = 2**args.num_bits
     max_q = torch.tensor(bit_range / 2 - 1, device=x.device)
     min_q = torch.tensor(-bit_range / 2, device=x.device)
-    Q = torch.zeros_like(x)
-    Q = quantize(x, scale, zero_point, min_q, max_q)
-    return dequantize(Q, scale, zero_point)
+    # Q = torch.zeros_like(x)
+    DQ = torch.zeros_like(x)
+    num_groups = len(scale)
+    group_size = int(x.shape[1] / num_groups)
+    for i in range(num_groups):
```

Comment:
can we share the logic for iterating over groups for a tensor of a certain size between both usages?